packages/atomic/src/components/facets/facet-search/facet-search.tsx

```diff
@@ -1,121 +1,116 @@
 import {h} from '@stencil/core';
-
 import ClearIcon from 'coveo-styleguide/resources/icons/svg/clear.svg';
 import {Combobox} from '../../../utils/combobox';
-import {I18nState} from '../../../utils/initialization-utils';
+import {Bindings, I18nState} from '../../../utils/initialization-utils';
 import {
   CategoryFacet,
-  CategoryFacetState,
   Facet,
-  FacetState,
+  SpecificFacetSearchResult,
+  CategoryFacetSearchResult,
 } from '@coveo/headless';
-import {BaseFacetSearchResult} from '@coveo/headless/dist/api/search/facet-search/base/base-facet-search-response';
 import {randomID} from '../../../utils/utils';
-import {CategoryFacetSearchResult} from '@coveo/headless/dist/api/search/facet-search/category-facet-search/category-facet-search-response';
 import SearchIcon from 'coveo-styleguide/resources/icons/svg/search.svg';
 import {sanitize} from '../../../utils/xss-utils';
 import {regexEncode} from '../../../utils/string-utils';
 
-export interface FacetSearchState {
+type FacetSearchResult = SpecificFacetSearchResult &
+  Partial<Pick<CategoryFacetSearchResult, 'path'>>;
+
+const SEPARATOR = '/';
+const ELLIPSIS = '...';
+const PATH_MAX_LENGTH = 3;
+
+export interface FacetSearchComponent {
+  bindings: Bindings;
   strings: I18nState;
   facetSearchQuery: string;
   showFacetSearchResults: boolean;
   facet: Facet | CategoryFacet;
-  facetState: FacetState | CategoryFacetState;
-}
-
-export class FacetSearchController {
-  constructor(public state: FacetSearchState) {}
-
-  public get facetSearch() {
-    return this.state.facet.facetSearch;
-  }
-
-  public get facetSearchState() {
-    return this.state.facetState.facetSearch;
-  }
-
-  public get strings() {
-    return this.state.strings;
-  }
-
-  public set text(text: string) {
-    this.state.facetSearchQuery = text;
-    this.facetSearch.updateText(text);
-    this.facetSearch.search();
-  }
 }
 
-export type FacetSearchProps = {
-  controller: FacetSearchController;
-};
-
 export class FacetSearch {
   private static ShowMoreResultsValue = -1;
   private inputRef!: HTMLInputElement;
   private valuesRef!: HTMLElement;
   private containerRef!: HTMLElement;
   private combobox: Combobox;
 
-  constructor(private props: FacetSearchProps) {
+  constructor(public component: FacetSearchComponent) {
     this.combobox = new Combobox({
       id: randomID('facet-search'),
-      strings: props.controller.state.strings,
+      strings: this.component.strings,
       containerRef: () => this.containerRef,
       inputRef: () => this.inputRef,
       valuesRef: () => this.valuesRef,
       onChange: (value) => {
-        this.props.controller.state.showFacetSearchResults = true;
-        this.props.controller.text = value;
+        this.component.showFacetSearchResults = true;
+        this.text = value;
       },
-      onSubmit: () => props.controller.facetSearch.search(),
+      onSubmit: () => {},
```

Comment:
Why is this now empty ?

Reply:
it updates the text and triggers a query onFocus and onChange already, this was not useful for the component